### PR TITLE
Diva-55 Require attribute

### PR DESCRIPTION
**Pull request by koen92**

### CODE REVIEW

File: src/diva.js [hunk 1]
```diff
@@ -32,24 +32,51 @@ function init(options) {
...
-    divaState.get(divaSessionId).attributes !== undefined &&
-  divaState.get(divaSessionId).attributes[attribute] !== undefined;
+function mergeAttribute(attributes, attributeName, attributeValue) {
+  const mergedAttributes = attributes;
+  if (attributes[attributeName] === undefined) {
+    mergedAttributes[attributeName] = [attributeValue];
+  } else {
+    mergedAttributes[attributeName] = attributes[attributeName].push(attributeValue);
+  }
+  return mergedAttributes;
```
Comment (timenolthof) [inline src/diva.js]: I would write this as:
```
return attributes[attributeName] ? {
  ...attributes,
  [attributeName]: attributes[attributeName].push(attributeValue),
} : {
 ...attributes,
  [attributeName]: attributeValue,
}
```

File: src/diva.js [hunk 2]
```diff
@@ -32,24 +32,51 @@ function init(options) {
...
 }
 
-function requireAttribute(attribute) {
+function getAttributes(divaSessionId) {
+  if (divaState[divaSessionId] === undefined) {
+    return {};
+  }
+
+  let attributes = {};
+  Object.values(divaState[divaSessionId]).forEach((proof) => {
```
Comment (timenolthof) [inline src/diva.js:51]: this might be easier with `Object.values().reduce`

File: src/diva.js [hunk 3]
```diff
@@ -62,9 +89,16 @@ function updateQRContentWithApiEndpoint(qrContent) {
   };
 }
 
+function attributesToContent(attributes, attributesLabel) {
```
Comment (timenolthof) [inline src/diva.js:92]: I think we should just require a syntax similar to what the `content` format requires.
We can then also supply convenience methods for simple cases.

The alternative is that we suplly a kind of DSL/builder syntax but that seems overkill to me.

**Review by timenolthof — APPROVED** [review]
Approved, but see the code simplification suggestions.

File: src/diva.js (hunk 3 above)
Comment (koen92) [reply src/diva.js:92]: will do in DIVA-58

**Pull request merged**